### PR TITLE
feat: Use Apache Commons ZipFile instead of the java.util one

### PR DESCRIPTION
**Pull request by aababilov**
java.util.zip supports two kinds of ZIP:
1) ZipFile - files on local disk
2) ZipInputStream - InputStream that may be read once

Our ZIP files may be remote (e.g., URLs), so option (1) does not work.
Some ZIP files may not be read by (2) - see a similar problem for Ruby
Zip::InputStream at
https://github.com/rubyzip/rubyzip#notice-about-zipinputstream

That is why we switch to Apache Commons ZipFile that supports reading
random-access streams from memory.

GtfsInputTest was changed to use TemporaryFolder instead of creating and
removing temporary files explicitly. Now we won't have a garbage
`core/storage` file at the git repository anymore.

### CODE REVIEW

**Review by lionel-nj — COMMENTED** [review]
Thanks @aababilov. One question in the comments. 


> GtfsInputTest was changed to use TemporaryFolder instead of creating and
> removing temporary files explicitly. Now we won't have a garbage
> core/storage file at the git repository anymore.

Thanks!

File: core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsZipFileInput.java [hunk 1]
```diff
@@ -55,15 +57,24 @@ static boolean isInsideZipDirectory(ZipEntry entry) {
 
   @Override
   public Set<String> getFilenames() {
-    return Collections.unmodifiableSet(filenames);
+    return filenames;
```
[the remark this answers is not in the document]
Comment (aababilov) [reply core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsZipFileInput.java:63]: OK, let's return an immutable one.

I can change the signature of this function to return ImmutableSet in a separate PR.

**Comment by aababilov** [comment]
Thanks, PTAL!

**Review by lionel-nj — APPROVED** [review]
Thanks @aababilov for providing this improvement

**Pull request merged**